tests/modules/test_text_processor.py

```diff
@@ -260,6 +274,139 @@ def test_subscribe_twice_hindi(self, texting_mock, logging_info_mock, logging_er
+        first_response = t.process(hindi_remind() + " SANJIV 25-11-2012")
+        self.assertEqual(first_response, msg_subscribe("Hindi").format(name="Sanjiv"))
+        texting_mock.assert_called_with(message=first_response, phone_number="1-111-1115")
+        self.assertTrue(Contact.objects.filter(name="Sanjiv", phone_number="1-111-1115", preg_update=False).exists())
+        self.assertTrue(t.get_contacts().exists())
+        t2 = TextProcessor(phone_number="1-111-1115")
+        self.assertTrue(t2.get_contacts().exists())
+        second_response = t2.process(hindi_remind() + " SANJIV 25-11-2012")
+        self.assertEqual(second_response, msg_already_sub("Hindi"))
+        texting_mock.assert_called_with(message=second_response, phone_number="1-111-1115")
+        logging_error_mock.assert_called_with("Contact for Sanjiv at 1-111-1115 was subscribed but already exists!")
+        contacts = Contact.objects.filter(name="Sanjiv", phone_number="1-111-1115", preg_update=False)
+        self.assertTrue(contacts.exists())
+        self.assertEqual(contacts.count(), 1)
+        self.assertTrue(t2.get_contacts().exists())
+
+    @patch("modules.text_processor.Texter.send")
+    def test_text_in_pregnancy_birthdate_update_english(self, texting_mock):
+        t = TextProcessor(phone_number="1-111-1114")
+        self.assertFalse(Contact.objects.filter(name="Rose", phone_number="1-111-1114").exists())
+        first_response = t.process("JOIN ROSE 25-10-2012")
+        self.assertEqual(first_response, msg_subscribe("English").format(name="Rose"))
+        texting_mock.assert_called_with(message=first_response, phone_number="1-111-1114")
+        self.assertTrue(Contact.objects.filter(name="Rose", phone_number="1-111-1114",date_of_birth=datetime(2012, 10, 25, 0, 0), preg_update=False).exists())
+        self.assertTrue(t.get_contacts().exists())
+        t2 = TextProcessor(phone_number="1-111-1114")
+        second_response = t2.process("Born ROSE 25-11-2012")
+        self.assertTrue(t2.get_contacts().exists())
+        self.assertEqual(second_response, msg_subscribe("English").format(name="Rose"))
+        texting_mock.assert_called_with(message=second_response, phone_number="1-111-1114")
+        self.assertFalse(Contact.objects.filter(name="Rose", phone_number="1-111-1114",date_of_birth=datetime(2012, 10, 25, 0, 0), preg_update=False).exists())
+        self.assertFalse(Contact.objects.filter(name="Rose", phone_number="1-111-1114",date_of_birth=datetime(2012, 11, 25, 0, 0), preg_update=False).exists())
+        self.assertTrue(Contact.objects.filter(name="Rose", phone_number="1-111-1114", date_of_birth=datetime(2012, 11, 25, 0, 0), preg_update=True).exists())
+        self.assertTrue(t2.get_contacts().exists())
+        actual_groups = [str(g) for g in t2.get_contacts().first().group_set.all()]
+        expected_groups = ['Everyone - English', 'Text Sign Ups', 'Text Sign Ups - English']
+        self.assertEqual(actual_groups, expected_groups)
```

Comment:
Need to test that the preg update happens.

Reply:
Line 332 tests that. Lines 330 and 331 preceding it check that it's the same same contact.

Reply:
Gotcha.

Reply:
Wasn't used to testing this by searching for `preg_update` via `filter` but I see now.